File: java/0396-rotate-function.java
class Solution {
    public int maxRotateFunction(int[] nums) {
        int n = nums.length;
        int totalSum = 0;
        int F = 0;
        for (int i = 0; i < n; i++) {
            totalSum += nums[i];
            F += i * nums[i];
        }
        int max = F;
        for (int k = 1; k < n; k++) {
            int rotatedOutIndex = n - k;
            F = F + totalSum - n * nums[rotatedOutIndex];
            max = Math.max(max, F);
        }
        return max;
    }
}

File: python/0396-rotate-function.py
class Solution:
    def maxRotateFunction(self, nums):
        n = len(nums)
        total_sum = sum(nums)
        F = sum(i * num for i, num in enumerate(nums))
        result = F
        for k in range(1, n):
            F = F + total_sum - n * nums[n - k]
            result = max(result, F)
        return result